test/onnx/test_pytorch_onnx_caffe2.py

```diff
@@ -973,7 +973,7 @@ class ComparisonModel(torch.nn.Module):
             def forward(self, x, y):
                 return x.ge(0.5) & y.le(2)
 
-        x = torch.ones(2, 3, dtype=torch.int32)
+        x = torch.ones(2, 3, dtype=torch.double)
```

Comment:
the `int32` type is intended, to test exporter's support for pytorch's implicit casting.

In exported onnx model, `x` will be promoted to float, with inserted cast node.

Reply:
Just to confirm: was the cast thing done after this PR was proposed? IIRC, I thought my old PR (https://github.com/onnx/onnx/pull/2783) made the test fail because int32 was not supported here.

Reply:
good point, checked the commit history of the implicit cast support. It does appear that the pass itself exists before your old PR, however there seems to be a bug with comparison ops, and was later fixed by 55ac2405893165de0687960a6650bbc5c8f29e73.